dask/dataframe/io/io.py

```diff
@@ -835,7 +855,10 @@ def chunksizes(ind):
     i = chunksizes(0)
     ind = None  # ind cache (sometimes avoids nonzero call)
     drift = 0  # accumulated drift away from ideal chunksizes
-    divs_remain = npartitions - len(divisions) if enforce_exact else None
+    divs_remain = None
+    if enforce_exact:
+        assert isinstance(npartitions, int)
```

Comment:
Why bother with these asserts? We would simply error on the next line; and there is no additional information passed to the caller here.

Reply:
That is because `mypy` otherwise gives errors. The simplest thing would be to remove annotations from the signature of those methods, and then `assert`s can be removed. But ideally all methods should be property annotated...

Reply:
It should be protected by a if `TYPE_CHECKING` block in that case

Reply:
How's that used? I'm not familiar with this `TYPE_CHECKING` block.

Reply:
Like https://github.com/dask/dask/blob/345a3822b561d9d7ef83fc276d3810adba6855fc/dask/dataframe/io/io.py#L40 
This convention is fairly standard and shows the intent of the assert - and will never make an assertion error when normally running the code.